src/utils/embedded/embedded.utils.ts

```diff
@@ -205,6 +205,50 @@ async function insecurelyValidateApplication() {
 // Validate immediately on load
 // insecurelyValidateApplication();
 
+const getAbsoluteUrl = (url: string): string | null => {
+  if (!url) {
+    return null;
+  }
+  try {
+    return new URL(url, window.location.origin).href;
+  } catch {
+    return null;
+  }
+};
+
+export function getAppLogo(): string | null {
+  try {
+    const faviconSelectors = [
+      "link[rel='icon'][type='image/svg+xml']",
+      "link[rel='icon'][sizes='32x32']",
+      "link[rel='icon'][sizes='48x48']",
+      "link[rel='icon'][sizes='96x96']",
+      "link[rel='icon']",
+      "link[rel='shortcut icon']",
+      "link[rel='apple-touch-icon'][sizes='180x180']",
+      "link[rel='apple-touch-icon']",
+      "link[rel='apple-touch-icon-precomposed']",
+      "meta[property='og:image']",
```

Comment:
`og:image` is usually a rather large image meant to be displayed as a preview when sharing a link, so I would avoid using it as icon.